spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -389,24 +389,19 @@ def compute_sql_info(
 
                         for task in stage.tasks:
                             sql_tasks.append(task.task_id)
-
-            df = df.append(
-                pd.DataFrame.from_dict(
-                    {
-                        "sql_id": [sqlid],
-                        "description": sql["description"],
-                        "start_time": [sql["start_time"] - app_model.start_time],
-                        "end_time": [sql["end_time"] - app_model.start_time],
-                        "duration": [sql["end_time"] - sql["start_time"]],
-                        "job_ids": [sql_jobs],
-                        "stage_ids": [sql_stages],
-                        "task_ids": [sql_tasks],
-                    }
-                )
-            )
+            dfs.append(pd.DataFrame.from_dict({
+                "sql_id": [sqlid],
+                "description": sql["description"],
+                "start_time": [sql["start_time"] - app_model.start_time],
+                "end_time": [sql["end_time"] - app_model.start_time],
+                "duration": [sql["end_time"] - sql["start_time"]],
+                "job_ids": [sql_jobs],
+                "stage_ids": [sql_stages],
+                "task_ids": [sql_tasks],
+            }))
```

Comment:
I ended up bumping the version of `pandas` we use as well, which resulted in some deprecation warnings about `DataFrame.append`, and that `pandas.concat` should be used instead (see - https://pandas.pydata.org/docs/whatsnew/v1.4.0.html#whatsnew-140-deprecations-frame-series-append). So I just changed this to collecting the smaller DataFrames into a list, and then concat-ing them together at the end.